client/src/views/workspaces/WorkspaceSharingModal.vue

```diff
@@ -310,6 +454,71 @@ async function updateUserRole(
   }
   await refreshSharingInfo();
 }
+
+async function onBatchRoleChange(newRoleOption: MsOption): Promise<void> {
+  let changesMade: boolean = false;
+  let errorOccured: boolean = false;
+
+  for (const userRole of selectedUsers.value) {
+    if (!userRole.role || newRoleOption.key !== userRole.role) {
+      const result = await shareWorkspace(props.workspaceId, userRole.user.id, newRoleOption.key);
+      if (result.ok) {
+        changesMade = true;
+      } else {
+        errorOccured = true;
+      }
+    }
+  }
+
+  if (!changesMade && !errorOccured) {
+    props.informationManager.present(
+      new Information({
+        message: {
+          key: 'WorkspaceSharing.batchSharing.noChanges',
+          data: {
+            role: I18n.translate(getWorkspaceRoleTranslationKey(newRoleOption.key).label),
+          },
+        },
+        level: InformationLevel.Info,
+      }),
+      PresentationMode.Toast,
+    );
+  } else if (errorOccured) {
```

Comment:
It's still not enough. If error is set, it could mean that all failed, or that only some failed. It's important to make the distinction, and in case they all failed, to have a bit more information (I mentioned storing the first error).

Replacing `changesMade` and `errorOccured` by counters would make this way more easy, and you'd also be able to compare to `selectedUsers.value.count` to know how many operation were planned, how many failed, how many succeeded.